src/client/pages/index/waterfall.vue

```diff
@@ -0,0 +1,285 @@
+        result.push({
+          name: typeItem ? typeItem.name : id,
+          value: [index, s.start, (s.end - s.start) < 1 ? 1 : s.end, duration],
+          itemStyle: {
+            normal: {
+              color: typeItem ? typeItem.color : '#000',
+            },
+          },
+        })
+      }
+    })
+  })
+
+  // console.log(result)
+
+  return result
+})
+
+async function refetch() {
+  data.value = await rpc.getWaterfallInfo(inspectSSR.value)
+}
+
+onRefetch.on(refetch)
+watch(inspectSSR, refetch)
+
+getHot().then((hot) => {
+  if (hot) {
+    hot.on('vite-plugin-inspect:update', refetch)
+  }
+})
+
+use([
+  VisualMapComponent,
+  CanvasRenderer,
+  BarChart,
+  TooltipComponent,
+  TitleComponent,
+  LegendComponent,
+  GridComponent,
+  DataZoomComponent,
+  CustomChart,
+])
+
+function renderItem(params: CustomSeriesRenderItemParams | any, api: CustomSeriesRenderItemAPI): CustomSeriesRenderItemReturn {
+  const categoryIndex = api.value(0)
+  const start = api.coord([api.value(1), categoryIndex])
+  const end = api.coord([api.value(2), categoryIndex])
+  const height = (api.size?.([0, 1]) as number[])[1] * 0.6
+  const rectShape = graphic.clipRectByRect(
+    {
+      x: start[0],
+      y: start[1] - height / 2,
+      width: end[0] - start[0],
+      height,
+    },
+    {
+      x: params.coordSys.x,
+      y: params.coordSys.y,
+      width: params.coordSys.width,
+      height: params.coordSys.height,
+    },
+  )
+
+  return (
+    rectShape && {
+      type: 'rect',
+      transition: ['shape'],
+      shape: rectShape,
+      style: api.style(),
+    }
+  )
+}
+
+const option = computed(() => ({
+  tooltip: {
+    formatter(params: TopLevelFormatterParams | any) {
+      return `${params.marker + params.name}: ${params.value[3] <= 1 ? '<1' : params.value[3]}ms}`
+    },
+
+  } satisfies TooltipComponentOption,
+  legendData: {
+    top: 'center',
+    data: ['c'],
+  } satisfies LegendComponentOption,
+
+  title: {
+    text: 'Waterfall',
+    // left: 'center',
+  },
+  visualMap: {
+    type: 'piecewise',
+    // show: false,
+    orient: 'horizontal',
+    left: 'center',
+    bottom: 10,
+    pieces: [
+
+    ],
+    seriesIndex: 1,
+    dimension: 1,
+  },
+  dataZoom: [
+    // 最多支持放大到1ms
+
+    {
+      type: 'slider',
+      filterMode: 'weakFilter',
+      showDataShadow: false,
+      top: height.value - dataZoomBar,
+      labelFormatter: '',
+    },
+    {
+      type: 'inside',
+      filterMode: 'weakFilter',
+    },
+  ],
+  grid: {
+    height: height.value - dataZoomBar - zoomBarOffset,
+  },
+  xAxis: {
+    min: startTime.value,
+    max: endTime.value,
+    // type: 'value',
+
+    scale: true,
+    axisLabel: {
+      formatter(val: number) {
+        return `${(val - startTime.value).toFixed(val % 1 ? 2 : 0)} ms`
+      },
+    },
+  } satisfies SingleAxisComponentOption,
+  yAxis: {
+    data: categories.value,
+  } satisfies SingleAxisComponentOption,
+  series: [
```

Comment:
group by id or plugin name ?
```suggestion
  series: [
```